### ISSUE TITLE
\todo ignores EXCLUDE_SYMBOLS

### ISSUE THREAD

**Issue by eseiler** [doxygen/doxygen#7738]
With the current doxygen release (`1.8.18 (a1b07ad0e92e4526c9ba1711d39f06b58c2a7459)`) I am encountering an unexpected behavior when mixing `\todo` with `EXCLUDE_SYMBOLS`.

Consider the following code:
```cpp
namespace detail
{
    /*!\brief Description.
     * \todo Write me.
     */
    struct foo{};
}
```

Using the standard Doxyfile, doxygen works fine and the documentation works as expected. The Todo page shows the todo description and links to the struct.

Setting `EXCLUDE_SYMBOLS  = detail` creates the warning `todo:1: warning: expected <dt> tag but found <p> instead!` and the Todo page shows only the description without any link.

As far as I understand, the `\todo` should be ignored since `detail::foo` is excluded?

Attached you find the failing example.
[report.tar.gz](https://github.com/doxygen/doxygen/files/4552505/report.tar.gz)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7740

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
